File: Problem_Solving_Algorithms/simple_array_sum.rb
# Using .each method

def simpleArraySum(array)
    sum = 0
    array.each do |number|
        sum += number
    end
    sum
end

# Using .reduce method

def simpleArraySum(array)
    array.reduce(0) { |sum, num| sum + num }
end

# Using .reduce method - shortcut

def simpleArraySum(array)
    array.reduce(:+)
end

File: Problem_Solving_Algorithms/JavaScript/simple_array_sum.js
// Using .reduce method

function simpleArraySum(array) {
    return array.reduce((nextValue, currentValue) => nextValue + currentValue);
}

// Using "for"

function simpleArraySum(array) {
    let sum = 0
    for (let i = 0; i < array.length; i++) {
        sum += array[i];
    }
    return sum;
}

// Using .forEach method

function simpleArraySum(array) {
    let sum = 0
    array.forEach(element => {
        sum += element;
    });
    return sum;
}
